### PR TITLE
Move dns resolution function from net to dns

### PR DESCRIPTION
**Pull request by AntonioLangiu**
As described in the issue #988

### CODE REVIEW

**Comment by AntonioLangiu** [comment]
@bassosimone I moved different parts of code from net/connect to dns/query. I'm going to modify them according to the issue #988, do you prefer if I do it here or if I open another sequential PR based on this?

File: src/libmeasurement_kit/net/connect.hpp [hunk 1]
```diff
@@ -31,13 +31,6 @@ void connect_first_of(Var<ConnectResult> result, int port,
                       Var<Logger> logger = Logger::global(), size_t index = 0,
                       Var<std::vector<Error>> errors = nullptr);
 
-typedef std::function<void(ResolveHostnameResult)> ResolveHostnameCb;
```
Comment (AntonioLangiu) [inline src/libmeasurement_kit/net/connect.hpp]: This is the only part of code I removed when moving code. I removed it in favour of `Callback<ResolveHostnameResult> cb`

**Comment by bassosimone** [comment]
At a first glance it seems good to me and I'd like to include it in 0.5.0-alpha. While at it, can you update the documentation and eventually fix other documentation inconsistencies and make sure the moved piece of code has coverage 100% (possibly it would be great to achieve coverage 100% *without* integration tests).

**Comment by bassosimone** [comment]
@AntonioLangiu, can you document the functions and structures made public by this PR?

**Comment by AntonioLangiu** [comment]
> update the documentation and eventually fix other documentation inconsistencies

I updated the documentation but I will take care of it after the next PR, that is going to refactor this code using **effectively the system resolver** and fixing some issues.

> make sure the moved piece of code has coverage 100%

It already has [100% coverage](https://coveralls.io/builds/9360981/source?filename=src%2Flibmeasurement_kit%2Fdns%2Fquery.cpp#L30). The case where the hostname is an ip (v4 or v6) address is tested with unit test. The case where the hostname is a domain is tested with integration tests, but it's a very used function and it's tested also by all the other procedures using it (consider that the resolution part is executed 699 times).

**Pull request merged**